### ISSUE TITLE
[DataGrid] Content overlapping when expanding row height while the detail panel is open.

### ISSUE THREAD

**Issue by Matthematic** [mui/mui-x#10704]
### The problem in depth 🔍

I am trying to design a datagrid in which each row has a cell that can expand/collapse and each row also has a master detail panel. Since my row has a dynamic height I set the getRowHeight to 'auto' and it works perfectly with the master detail panel closed, but once I open the master detail there is some strange overlapping happening. I expected the master detail to act like an injection of scrollheight and so it should move down at the same rate that the row is expanding, but it is slow to move and ends up displaying on top of the row content. The delay before it eventually moves to the correct position is quite long, and my assumption is that there is some resize debouncing going on for performance reasons. Is there something I can do to avoid this?

Video: 
![DataGrid bug](https://github.com/mui/mui-x/assets/6292111/c0639354-0c62-49b6-91d9-f97b1e735f62)

CodeSandbox:
https://codesandbox.io/s/zealous-bose-lg77fd?file=/src/demo.tsx

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>
  
```
System:
    OS: macOS 13.1
  Binaries:
    Node: 18.16.1 - ~/.nvm/versions/node/v18.16.1/bin/node
    Yarn: 3.6.1 - ~/.nvm/versions/node/v18.16.1/bin/yarn
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.1/bin/npm
  Browsers:
    Chrome: 118.0.5993.70
    Edge: Not Found
    Safari: 16.2
  npmPackages:
    @emotion/react: ^11.11.1 => 11.11.1 
    @emotion/styled: ^11.11.0 => 11.11.0 
    @mui/base:  5.0.0-beta.14 
    @mui/core-downloads-tracker:  5.14.8 
    @mui/icons-material: ^5.11.16 => 5.11.16 
    @mui/material: ^5.13.6 => 5.13.6 
    @mui/private-theming:  5.14.8 
    @mui/styled-engine:  5.14.8 
    @mui/system: ^5.14.1 => 5.14.1 
    @mui/types:  7.2.4 
    @mui/utils:  5.14.8 
    @mui/x-data-grid-generator: ^6.15.0 => 6.15.0 
    @mui/x-data-grid-pro: ^6.16.2 => 6.16.2 
    @mui/x-date-pickers-pro: ^6.15.0 => 6.15.0 
    @mui/x-license-pro: ^6.10.2 => 6.10.2 
    @types/react: ^18.0.37 => 18.0.37 
    react: ^18.2.0 => 18.2.0 
    react-dom: ^18.2.0 => 18.2.0 
    typescript: ^5.0.2 => 5.0.2 
```
</details>

Tested using Chrome 117.0.5938.92

**Search keywords**: datagrid row dynamic height master detail panel
**Order ID**: f2065eaa10dc464019fd61bf90df1bc5Tz03NDE3MCxFPTE3MjU2NTQ3MTUwMDAsUz1wcm8sTE09c3Vic2NyaXB0aW9uLEtWPTI=

**Comment by cherniavskii** [comment]
Hey @Matthematic 
Indeed, we debounce row position calculations for performance reasons.
We can make it configurable though.
I'll submit a PR 👍🏻

**Comment by romgrk** [comment]
@cherniavskii This is fixed by #10059 because the detail panels are displayed in the same container as the rows after that PR, so height transition pushes the panel smoothly. Not sure if you want to merge your PR or let sticky fix this.

**Comment by cherniavskii** [comment]
@romgrk Nice, this PR is solving so many issues! 🎉
I'm going to proceed with the PR anyway to have a workaround for this issue in v6
